esb/plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/AbstractMediator.java

```diff
@@ -190,6 +241,35 @@ public void activate() {
 		if (!reversed) {
 			reverse(this);
 		}
+		if (i == 0) {
+			if (ESBDebuggerUtil.getRecentlyAddedMediator() == null
+					&& (!ESBDebuggerUtil.ispageSaveOperationActivated()
+							&& !ESBDebuggerUtil
+									.isPageChangeOperationActivated() && !ESBDebuggerUtil
+								.isPageCreateOperationActivated())) {
+				ESBDebuggerUtil.setRecentlyAddedMediator(this);
+			} else {
+
+				try {
+					if (ESBDebuggerUtil.getRecentlyAddedMediator() != null) {
+						ESBDebuggerUtil
```

Comment:
These sections looks messy, break the code lines at proper places, to make it more readable.